### PR TITLE
Enable support fo EdDSA signatures in addition to DSA

### PR DESCRIPTION
**Pull request by kkyryl**
Both are supported by the App Center but only DSA is currently supported by the fastlane plugin

### CODE REVIEW

File: lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb [hunk 1]
```diff
@@ -546,15 +546,23 @@ def self.update_release(api_token, owner_name, app_name, release_id, release_not
...
+      def self.update_release_metadata(api_token, owner_name, app_name, release_id, dsa_signature = '', ed_signature = '')
+        return if dsa_signature.to_s == '' && ed_signature.to_s == ''
 
         url = "v0.1/apps/#{owner_name}/#{app_name}/releases/#{release_id}"
-        body = {
-          metadata: {
-            dsa_signature: dsa_signature
-          }
-        }
+        if dsa_signature.to_s != ''
```
Comment (kkyryl) [inline lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb:553]: In my experience, those are not used together therefore this check is either/or.

Comment (jp-andre) [reply lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb:553]: Agreed. You can enforce that with `conflicting_options` and `conflict_block` in the Options list. Or alternatively allow passing both like #222 was doing.

File: lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb [hunk 2]
```diff
@@ -143,6 +143,7 @@ def self.run_release_upload(params)
         build_number = params[:build_number]
         version = params[:version]
         dsa_signature = params[:dsa_signature]
+        ed_signature = params[:ed_signature]
```
Comment (jp-andre) [inline lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb:146]: `eddsa` would be easier to understand in my opinion.

**Review by jp-andre — COMMENTED** [review]
Looks good but this is lacking tests (specs). Would love to merge if we get new tests.
Also we need to update README to add the new field.

File: lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb (hunk 2 above)
Comment (kkyryl) [reply lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb:146]: I agree - but I was following the naming convention in the API and in the API the field is ed_signature. I can go either way, wasn't sure what's community preference here.

File: lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb (hunk 1 above)
Comment (kkyryl) [reply lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb:557]: @jp-andre Updated to allow both values.

**Comment by kkyryl** [comment]
@jp-andre Updated per your comments:
- Added support for both values
- Updated README
- Updated spec

**Comment by kkyryl** [comment]
@jp-andre any chance we can move this forward?

File: lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb [hunk 3]
```diff
@@ -546,15 +546,20 @@ def self.update_release(api_token, owner_name, app_name, release_id, release_not
...
-          }
+          metadata: {}
         }
+        
+        if dsa_signature.to_s != ''
+          body["metadata"]["dsa_signature"] = dsa_signature
+        end
+        if ed_signature.to_s != ''
+          body["metadata"]["ed_signature"] = ed_signature
+        end
```
Comment (jp-andre) [inline lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb:562]: This code is not valid. `metadata` in `body` is a symbol (`:metadata`), you're indexing here as a string `"metadata"`.
```suggestion
        if dsa_signature.to_s != ''
          body[:metadata]["dsa_signature"] = dsa_signature
        end
        if ed_signature.to_s != ''
          body[:metadata]["ed_signature"] = ed_signature
        end
```

**Review by jp-andre — CHANGES_REQUESTED** [review]
Sorry, the spec tests don't pass. I've added a change (how was I allowed to write in your fork? I wonder). Please kindly fix the tests and ping me again if I forget about this PR.

**Comment by kkyryl** [comment]
@jp-andre thank you for helping with the code, appreciate it. I fixed the issue with the spec and adjusted, looks like it passes now.

**Comment by kkyryl** [comment]
@jp-andre just pinging to see if we can get this going

**Review by jp-andre — APPROVED** [review]

**Review by anywherepilot — APPROVED** [review]

**Pull request merged**